app/services/feeds/hacker_news/top_stories.rb

```diff
@@ -0,0 +1,30 @@
+module Feeds
+  module HackerNews
+    # Returns an Array of ids for 500 of the top stories.
+    class TopStories < BaseService
+      include Limitable
+      include Serviceable
+      include Sourceable
+
+      def initialize(limit:)
+        @limit = limit
+
+        validate_limit!
+
+        super URI("#{BASE_URI}/v0/topstories.json")
+      end
+
+      def execute
+        response = request
+        unless response.status == 200
+          raise "Error calling '#{SOURCE}' service: http status: #{response.status}"
+        end
+
+        ids = JSON.parse(response.body)
+        return ids unless greater_than_limit? ids
+
+        ids.sample(limit)
+      end
+    end
+  end
+end
```

Comment:
DRY, uses inheritance. The Hacker news service that retrieves the top news story ids. These story ids are later used by another Hacker News service to retrieve the story details that populate our feeds table in our database.